self-tests.yml

```diff
@@ -585,6 +596,22 @@ tasks:
             LOCAL_PATH: clients
             EXCLUDE_PATTERN: .cache
             REMOTE_PATH: evergreen/clients/${project}/${revision}
+  - name: swaggo-lint
+    commands:
+      - func: get-project-and-modules
+      - func: get-swaggo
+      - command: shell.exec
+        params:
+          shell: bash
+          working_dir: evergreen
+          script: |
+            ./swag fmt -g service/service.go
+            if git diff --exit-code --diff-filter=M; then
```

Comment:
This should probably use the same git diffing logic as in the `scripts/verify-` scripts because of differences between local env vs. different types of patches. Manual patches apply the diff differently from GitHub PRs. I remember there were a bunch of issues with consolidating local git repo state vs. manual patch assumptions vs. GitHub PR checks when I added the first diff-based lint script.